firepit/clickhousestorage.py

```diff
@@ -0,0 +1,585 @@
+import logging
+import os
+import re
+import time
+import traceback
+
+import orjson
+import psycopg2
+import psycopg2.extras
+
+from firepit.exceptions import IncompatibleType
+from firepit.exceptions import InvalidAttr
+from firepit.exceptions import UnknownViewname
+from firepit.splitter import SqlWriter
+from firepit.sqlstorage import SqlStorage
+from firepit.sqlstorage import validate_name
+from firepit.query import Table
+from firepit.query import Join
+import jaydebeapi
+
+from firepit.stix20 import stix2sql
+
+from firepit.props import auto_agg
+from firepit.props import primary_prop
+from firepit.validate import validate_name
+from firepit.validate import validate_path
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_storage(url, session_id):
+    dbname = url.path.lstrip('/')
+    return ClickhouseStorage(dbname, url.geturl(), session_id)
+
+
+def _infer_type(key, value):
+    if key == 'id':
+        rtype = 'String'
+    elif isinstance(value, bool):
+        rtype = 'UInt8'
+    elif isinstance(value, int):
+        rtype = 'Int32'
+    elif isinstance(value, float):
+        rtype = 'Float32'
+    elif isinstance(value, list):
+        rtype = 'String'
+    else:
+        rtype = 'String'
+    return rtype
+
+
+
+class ConnectionWrapper():
+    def __init__(self, connection):
+         self.connection = connection
+
+    def cursor(self):
+        cursor =self.connection.cursor()
+        return CursorWrapper(cursor);
+
+    def close(self):
+        if not self.connection._closed:
+            self.connection.close()
+
+    def commit(self):
+        self.connection.commit()
+
+    def _closed(self):
+        return self.connection._closed
+
+class CursorWrapper():
+    def __init__(self, cursor,defaultDB=None):
+         self.cursor = cursor
+         #self.defaultDB = defaultDB
```

Comment:
Looks like the defaultDB isn't used?  Can you remove it?

Reply:
Done